javascript/jQuerry.js

```diff
@@ -0,0 +1,225 @@
+    button = keys[i];
+    $(button).on('click',function() {
+      buttonValue = $(this).text();
+
+      if (buttonValue != "AC") {
+        $(clearButton).text("C")
+      }
+
+      if (buttonValue == "C") {
+        stack = [];
+        $(clearButton).text("AC")
+        resultValue = 0;
+      }
+
+      if(buttonValue == "+") {
+        switch(stack.length){
+          case 1:
+            stack.push("+");
+            break;
+          case 2: case 4:
+            stack.pop();
+            stack.push("+");
+            break;
+          case 3:
+            num2 = stack.pop();
+            op1 = stack.pop();
+            num1 = stack.pop();
+            stack.push(operation(num1, op1, num2));
+            stack.push("+");
+            resultValue = operation(num1, op1, num2);
+            break;
+          case 5:
+            num3 = stack.pop();
+            op2 = stack.pop();
+            num2 = stack.pop();
+            op1 = stack.pop();
+            num1 = stack.pop();
+
+            res = operation(num2, op2, num3);
+            resultValue = operation(num1, op1, res);
+            stack.push(resultValue);
+            stack.push("+");
+        }
+      }
+
+      if(buttonValue == "-") {
+        switch(stack.length){
+          case 1:
+            stack.push("-");
+            break;
+          case 2:
+            stack.pop();
+            stack.push("-");
+            break;
+          case 3:
+            num2 = stack.pop();
+            op1 = stack.pop();
+            num1 = stack.pop();
+            stack.push(operation(num1, op1, num2));
+            stack.push("-");
+            resultValue = operation(num1, op1, num2);
+            break;
+          case 5:
+            num3 = stack.pop();
+            op2 = stack.pop();
+            num2 = stack.pop();
+            op1 = stack.pop();
+            num1 = stack.pop();
+
+            res = operation(num2, op2, num3);
+            resultValue = operation(num1, op1, res);
+            stack.push(resultValue);
+            stack.push("-");
+        }
+      }
+
+      if(buttonValue == "x") {
+        switch(stack.length){
+          case 1:
+            stack.push("x");
+            break;
+          case 2: case 4:
+            stack.pop();
+            stack.push("*");
+            break;
+          case 3:
+            if(stack[1]=="x" || stack[1]=="/") {
+              num2 = stack.pop();
+              op1 = stack.pop();
+              num1 = stack.pop();
+              stack.push(operation(num1, op1, num2));
+              stack.push("x");
+              resultValue = operation(num1, op1, num2);
+            }
+            else
+            {
+              stack.push("x");
+            }
+            break;
+          case 5:
+            num2 = stack.pop();
+            op1 = stack.pop();
+            num1 = stack.pop();
+            stack.push(operation(num1, op1, num2));
+            stack.push("x");
+            resultValue = operation(num1, op1, num2);
```

Comment:
Compute the resultValue first, then push it to the stack to avoid a function call